feat(ai-gateway: migrate agents and mcp (#5903)
Kong/developer.konghq.com: app/ai-gateway/migrate-agents.md

description: This guide walks you through moving your a2a agents to the new {{site.ai_gateway}} AI Agent entities.
---

In {{site.ai_gateway}} version 1.x, an agent is an [AI A2A Proxy](/plugins/ai-a2a-proxy/) plugin attached to a Service and Route. The plugin is a transparent proxy that adds observability and agent card URL rewriting to Agent-to-Agent (A2A) traffic (where the gateway automatically changes the agent's address so clients connect through the gateway instead of directly to the agent.)

> In {{site.ai_gateway}} version 2.x, that plugin becomes an [AI Agent](/ai-gateway/entities/ai-agent/) entity, which captures the following in a single entity and applies the agent card rewriting automatically:
> * The upstream URL
> * Routing
> * Logging

Reviewer: ```suggestion
In {{site.ai_gateway}} version 2.x, that plugin becomes an [AI Agent](/ai-gateway/entities/ai-agent/) entity, which captures the following in a single entity and applies the agent card rewriting automatically.
```

Author: That sentence would stop making sense if we remove the bullet list, because of "the following".

## Converting configuration files

> The following `deck` snippet defines an A2A agent that proxies to an upstream agent that handles flight bookings (the backend the gateway forwards requests to), with payload logging disabled:

Reviewer: ```suggestion
The following version 1.x example defines an A2A agent that proxies an upstream agent that handles flight bookings:
```

```
# kong.yaml (AI Gateway v1, exported with deck gateway dump)
services:
- name: flight-booking-agent
  url: https://booking-agent.internal.kongair.com
  routes:
  - name: flight-booking-agent-route
    paths:
    - /booking-agent
  plugins:
  - name: ai-a2a-proxy
    config:
      max_request_body_size: 8388608
      logging:
        log_statistics: true
        log_payloads: false
        max_payload_size: 1048576
```

> The following [converter](/ai-gateway/v2-migration-guide/#step-2-run-the-converter):

Reviewer: ```suggestion
Converting the example to use the version 2.x model:
```

* Moves the upstream URL, route, request-size limit, and logging settings onto a single AI Agent entity.
* Renames the `config.logging` fields: `log_statistics` becomes `statistics`, and `log_payloads` becomes `payloads`

> * Automatically applies the agent card rewriting

Reviewer: ```suggestion
```

Reviewer: this is unchanged between versions as noted in the 'what to check' section
